### ISSUE TITLE
Heartbeats dying from H3 Harvest Agent

### ISSUE THREAD

**Issue by obrienben** [DIA-NZ/webcurator#74]
It appears the heartbeat trigger is randomly dying and not resuming in the H3 Harvest Agent, only being fixed by a restart.
Other interactions with the Harvest Agent can still occur once this state happens:

- new harvests can be started, and H3 will run them, but no status updates will come back once the heartbeats have stopped.
- completed harvests in H3 will be detected and the completed job scheduled task will still run in the Harvest Agent.

**Comment by obrienben** [comment]
Suspect the rescanJobDirectory call failed in HarvesterH3.getStatus(), and a null EngineResult object caused the null pointer exception on line 192

```
EngineResult engineResult = heritrix.rescanJobDirectory();
            h3job = heritrix.job(h3job.shortName).job;
            if (h3job != null) {
                status.setJobName(h3job.shortName);
                status.setStatus(h3job.crawlControllerState);
                String heritrixBuild = engineResult.engine.heritrixVersion;
```

**Comment by obrienben** [comment]
Fixed by #79
